spec/superform_spec.rb

```diff
@@ -96,4 +97,28 @@
       end
     end
   end
+
+  context "input behavior" do
```

Comment:
In theory, this spec is already covered by https://github.com/rubymonolith/blob/b82992ff7b471a182c32e0db8b5ad717255c63a2/spec/superform_spec.rb#L74-L80

I don't quite understand what I'm missing in this PR that's leading to the bug you're seeing, but I do believe there is a bug.

I'd like to see this particular test moved into the spec I linked to since it should be rare for folks to pass a `field_class` into a Superform.

Reply:
Here's an example of how I'd like the specs to look (look for the 🍌)

```ruby
  it "has correct DOM names" do
    Superform :user, object: user do |form|
      form.namespace(:name) do |name|
        name.field(:first).tap do |first|
          first.dom.tap do |dom|
            expect(dom.id).to eql("user_name_first")
            expect(dom.name).to eql("user[name][first]")
          end
          # 🍌
          expect(first).to be_a Superform::Rails::Form::Field
        end
      end
      form.field(:nicknames).collection do |field|
        # 🍌
        expect(field).to be_a Superform::Rails::Form::Field
        field.dom.tap do |dom|
          expect(dom.id).to match /user_nicknames_\d+/
          expect(dom.name).to eql("user[nicknames][]")
        end
      end
      form.collection(:addresses) do |address|
        address.field(:street).dom.tap do |dom|
          expect(dom.id).to match /user_addresses_\d_street+/
          expect(dom.name).to eql("user[addresses][][street]")
        end
        address.field(:city)
        address.field(:state)
      end
      form.namespace(:one).namespace(:two).namespace(:three).field(:four).dom.tap do |dom|
        expect(dom.id).to eql("user_one_two_three_four")
        expect(dom.name).to eql("user[one][two][three][four]")
      end
    end
  end
```

If you can roll the behavior you'd like to see into those specs, we can get this to a spot where its mergable.